src/Integration/Binding/BindingProcessImpl.cs

```diff
@@ -97,52 +97,78 @@ internal class BindingProcessImpl : IBindingProcess
-            var languageList = this.GetBindingLanguages();
+            var languageList = GetBindingLanguages();
 
             var languageCount = languageList.Count();
             int currentLanguage = 0;
-            progress?.Report(new FixedStepsProgress(Strings.DownloadingQualityProfileProgressMessage, currentLanguage, languageCount));
 
             foreach (var language in languageList)
             {
-                var serverLanguage = language.ServerLanguage;
+                currentLanguage++;
+
+                var progressMessage = string.Format(Strings.DownloadingQualityProfileProgressMessage, language.Name);
+                progress?.Report(new FixedStepsProgress(progressMessage, currentLanguage, languageCount));
 
-                // Download the quality profile for each language
-                var qualityProfileInfo = await Core.WebServiceHelper.SafeServiceCallAsync(() =>
-                    this.host.SonarQubeService.GetQualityProfileAsync(
-                        this.bindingArgs.ProjectKey, this.bindingArgs.Connection.Organization?.Key, serverLanguage, cancellationToken),
-                    this.host.Logger);
+                var qualityProfileInfo = await TryDownloadQualityProfileAsync(language, cancellationToken);
+                
                 if (qualityProfileInfo == null)
                 {
-                    this.host.Logger.WriteLine(string.Format(Strings.SubTextPaddingFormat,
-                       string.Format(Strings.CannotDownloadQualityProfileForLanguage, language.Name)));
-                    return false;
+                    continue; // skip to the next language
                 }
-                this.InternalState.QualityProfiles[language] = qualityProfileInfo;
+
+                InternalState.QualityProfiles[language] = qualityProfileInfo;
 
                 var bindingConfiguration = QueueWriteBindingInformation();
 
                 // Create the binding configuration for the language
-                var bindingConfig = await this.bindingConfigProvider.GetConfigurationAsync(qualityProfileInfo, language, bindingConfiguration, cancellationToken);
+                var bindingConfig = await bindingConfigProvider.GetConfigurationAsync(qualityProfileInfo, language, bindingConfiguration, cancellationToken);
                 if (bindingConfig == null)
                 {
-                    this.host.Logger.WriteLine(string.Format(Strings.SubTextPaddingFormat,
+                    host.Logger.WriteLine(string.Format(Strings.SubTextPaddingFormat,
                         string.Format(Strings.FailedToCreateBindingConfigForLanguage, language.Name)));
                     return false;
                 }
 
-                this.InternalState.BindingConfigs[language] = bindingConfig;
+                InternalState.BindingConfigs[language] = bindingConfig;
 
-                currentLanguage++;
-                progress?.Report(new FixedStepsProgress(string.Empty, currentLanguage, languageCount));
```

Comment:
Progress reporting moved to to before the download starts. Previously the message was blank, which wasn't very helpful. Effectively it meant the progress message just updated the progress bar. Now it shows a message too.